src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -1132,14 +1132,22 @@ public override Binder VisitXmlNameAttribute(XmlNameAttributeSyntax parent)
             /// </summary>
             private Binder GetParameterNameAttributeValueBinder(MemberDeclarationSyntax memberSyntax, Binder nextBinder)
             {
-                BaseMethodDeclarationSyntax baseMethodDeclSyntax = memberSyntax as BaseMethodDeclarationSyntax;
-                if ((object)baseMethodDeclSyntax != null && baseMethodDeclSyntax.ParameterList.ParameterCount > 0)
+                if (memberSyntax is BaseMethodDeclarationSyntax { ParameterList: { ParameterCount: > 0 } } baseMethodDeclSyntax)
                 {
                     Binder outerBinder = VisitCore(memberSyntax.Parent);
                     MethodSymbol method = GetMethodSymbol(baseMethodDeclSyntax, outerBinder);
                     return new WithParametersBinder(method.Parameters, nextBinder);
                 }
 
+                if (memberSyntax is RecordDeclarationSyntax { ParameterList: { ParameterCount: > 0 } } recordDeclSyntax)
+                {
+                    Binder outerBinder = VisitCore(memberSyntax.Parent);
+                    SourceNamedTypeSymbol recordType = ((NamespaceOrTypeSymbol)outerBinder.ContainingMemberOrLambda).GetSourceTypeMember(recordDeclSyntax);
+                    var primaryConstructor = recordType.GetMembersUnordered().OfType<SynthesizedRecordConstructor>().SingleOrDefault();
```

Comment:
>primaryConstructor [](start = 24, length = 18)

I think this method might not correspond to the parameters in this partial declarartion.

Reply:
Added tests for partials.
We're basically taking the xml doc from the declaration that has the parameter list. That seems sensible. But we could also try to also find the xml doc on declaration without parameter list. What do you think?

---
In reply to: [517014473](https://github.com/dotnet/roslyn/pull/49134#discussion_r517014473) [](ancestors = 517014473)

Reply:
> We're basically taking the xml doc from the declaration that has the parameter list. That seems sensible. But we could also try to also find the xml doc on declaration without parameter list. What do you think?

I think that we should match doc-comment to the primary constructor only if both are on the same partial declaration

---
In reply to: [518320962](https://github.com/dotnet/roslyn/pull/49134#discussion_r518320962) [](ancestors = 518320962,517014473)

Reply:
Note that only one partial declaration declares a primary constructor, parameter lists on other declarations are ignored and not analyzed, can have all kind of differences by comparison to each other.

---
In reply to: [518379208](https://github.com/dotnet/roslyn/pull/49134#discussion_r518379208) [](ancestors = 518379208,518320962,517014473)

Reply:
Apparently my comment wasn't clear. Let me rephrase it. I think that here we should find primary constructor symbol only if it corresponds to memberSyntax. We shouldn't be using a symbol that is declared by a different partial declaration. 

---
In reply to: [518380246](https://github.com/dotnet/roslyn/pull/49134#discussion_r518380246) [](ancestors = 518380246,518379208,518320962,517014473)